okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -124,6 +126,14 @@ public void onResponse(Call call, Response response) throws IOException {
                         wrappedException.getStatus());
                 correctStackTrace.initCause(e);
                 throw correctStackTrace;
+            } else if (e.getCause() instanceof IoUnknownRemoteException) {
+                UnknownRemoteException wrappedException = ((IoUnknownRemoteException) e.getCause())
+                        .getWrappedException();
+                UnknownRemoteException correctStackTrace = new UnknownRemoteException(
+                        wrappedException.getStatus(),
+                        wrappedException.getBody());
+                correctStackTrace.initCause(e);
+                throw correctStackTrace;
```

Comment:
this throws a non-IOException. I'm not sure that's kosher given the Feign contract?

Reply:
But a RemoteException isn't an IOException either?

Reply:
oh, I misread, thanks